### PR TITLE
Fix sporadic `Table_Tests` failures caused by reliance on test ordering.

### PR DESCRIPTION
**Pull request by GregoryTravis**
It looks like the test "SQLite_Format should allow connecting to SQLite files" was responsible for cleaning up the SQLite test file, as it removed the file and was last in the test suite. Now that tests run out of order, this won't work.

This PR removes the teardown that was deleting the file early.

#9436 #9437

### Checklist

Please ensure that the following checklist has been satisfied before submitting the PR:

- [x] The documentation has been updated, if necessary.
- [x] Screenshots/screencasts have been attached, if there are any visual changes. For interactive or animated visual changes, a screencast is preferred.
- [x] All code follows the
      [Scala](https://github.com/enso-org/enso/blob/develop/docs/style-guide/scala.md),
      [Java](https://github.com/enso-org/enso/blob/develop/docs/style-guide/java.md),
      and
      [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md)
      style guides. In case you are using a language not listed above, follow the [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md) style guide.
- All code has been tested:
  - [x] Unit tests have been written where possible.
  - [x] If GUI codebase was changed, the GUI was tested when built using `./run ide build`.

### CODE REVIEW

File: test/Table_Tests/src/Common_Table_Operations/Select_Columns_Spec.enso [hunk 1]
```diff
@@ -539,7 +539,7 @@ add_specs suite_builder setup =
             t2.should_fail_with Ambiguous_Column_Rename
             err = t2.catch
             err.column_name . should_equal "beta"
-            err.new_names . should_equal ["FirstColumn", "DifferentName!"]
+            err.new_names.sort . should_equal ["DifferentName!", "FirstColumn"]
```
Comment (GregoryTravis) [inline test/Table_Tests/src/Common_Table_Operations/Select_Columns_Spec.enso:542]: This test probably relied on `Map` iteration order. (It failed when I changed the iteration order to reproduce the failures.)

File: test/Table_Tests/src/Common_Table_Operations/Select_Columns_Spec.enso [hunk 2]
```diff
@@ -587,7 +587,7 @@ add_specs suite_builder setup =
         group_builder.specify "should correctly handle problems: duplicate names" <|
             map = ["Test", "Test", "Test", "Test"]
             action = data.table.rename_columns map on_problems=_
-            tester = expect_column_names ["Test 1", "Test 2", "Test 3", "Test"]
+            tester = expect_column_names ["Test 1", "Test 2", "Test 3", "Test"] ignore_order=True
```
Comment (GregoryTravis) [inline test/Table_Tests/src/Common_Table_Operations/Select_Columns_Spec.enso:590]: This test probably relied on `Map` iteration order. (It failed when I changed the iteration order to reproduce the failures.)

File: test/Table_Tests/src/Database/SQLite_Spec.enso [hunk 3]
```diff
@@ -394,6 +394,9 @@ type File_Connection
 
 add_specs suite_builder =
     in_file_prefix = "[SQLite File] "
+    # TODO: Add a suite-level teardown to delete this file at the end.
```
Comment (GregoryTravis) [inline test/Table_Tests/src/Database/SQLite_Spec.enso:397]: This file is removed (if it exists) at the start of the test so it's okay if we don't clean it up for now.

File: test/Table_Tests/src/Database/SQLite_Spec.enso [hunk 4]
```diff
@@ -442,7 +442,7 @@ add_specs suite_builder =
         group_builder.specify "should connect to a db file" <|
             connection = Data.read data.file
             tables = connection.tables
-            tables.row_count . should_not_equal 0
```
Comment (GregoryTravis) [inline test/Table_Tests/src/Database/SQLite_Spec.enso:445]: I'm assuming here that if this test ever ran first, this line would have failed.

**Comment by Akirathan** [comment]
I see. So the `SQLite_Format should allow connecting to SQLite files` test group deletes the database file in its teardown. If other groups run after that, they will fail because the database file got deleted. Good observation. Note that it is not that necessary to delete that file at the end of testing, as it is deleted at the beginning of testing if necessary.

**Review by JaroslavTulach — APPROVED** [review]

**Review by radeusgd — APPROVED** [review]

**Pull request merged**